src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -585,4 +585,30 @@ public void ListenNamedPipe(string pipeName, Action<ListenOptions> configure)
         configure(listenOptions);
         CodeBackedListenOptions.Add(listenOptions);
     }
+
+    /// <summary>
+    /// Return true if there is a default or development server certificate.
+    /// Should not be called before the configuration is loaded, if there is one.
+    /// </summary>
+    /// <returns>True if there is a default or development server certificate, false otherwise.</returns>
+    /// <exception cref="InvalidOperationException">If there is a configuration and it has not been loaded.</exception>
+    public bool HasDefaultOrDevelopmentCertificate
+    {
+        get
+        {
+            if (ConfigurationLoader is { IsLoaded: false })
+            {
+                throw new InvalidOperationException(CoreStrings.NeedConfigurationToRetrieveServerCertificate);
+            }
+
+            // We consider calls to `HasDefaultOrDevelopmentCertificate.get` to be a clear expression of user intent to pull in HTTPS configuration support
+            EnableHttpsConfiguration();
+
+            var httpsOptions = new HttpsConnectionAdapterOptions();
+            ApplyHttpsDefaults(httpsOptions);
+            ApplyDefaultCertificate(httpsOptions);
+
+            return httpsOptions.ServerCertificate is not null;
```

Comment:
What happens if this method is called multiple times? Does it re-do all this work? Is it cached?

Reply:
I can't think of a scenario where it would make sense to call this more than once per host, but maybe we need to be defensive?  If so, we'd probably want to defend against the configuration changing between invocations, perhaps by tracking a configuration change token.

Reply:
Per our offline conversation, the question was about safety, rather than perf, and the method is safe to call repeatedly.